Sources/SwiftDocCUtilities/ArgumentParsing/Subcommands/DiffDocCArchive.swift

```diff
@@ -0,0 +1,246 @@
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import ArgumentParser
+import Foundation
+import SwiftDocC
+
+extension Docc.ProcessArchive {
+    
+    struct DiffDocCArchive: ParsableCommand {
+        
+        // MARK: - Content and Configuration
+        
+        /// Command line configuration.
+        static var configuration = CommandConfiguration(
+            commandName: "diff-docc-archive",
+            abstract: "Produce a markdown file saved as {FrameworkName}_ChangeLog.md containing the diff of added/removed symbols between the two provided DocC archives.",
+            shouldDisplay: true)
+        
+        /// Content of the 'changeLog' template.
+        static func changeLogTemplateFileContent(
+            frameworkName: String,
+            initialDocCArchiveVersion: String,
+            newerDocCArchiveVersion: String,
+            additionLinks: String,
+            removalLinks: String
+        ) -> [String : String] {
+            [
+                "\(frameworkName.localizedCapitalized)_ChangeLog.md": """
+                    # \(frameworkName.localizedCapitalized) Updates
+                    
+                    @Metadata { @PageColor(yellow) }
+                    
+                    Learn about important changes to \(frameworkName.localizedCapitalized).
+                    
+                    ## Overview
+
+                    Browse notable changes in \(frameworkName.localizedCapitalized).
+                    
+                    ## Version: Diff between \(initialDocCArchiveVersion) and \(newerDocCArchiveVersion)
+
+                    
+                    ### Change Log
+                    
+                    #### Additions
+                    _New symbols added in \(newerDocCArchiveVersion) that did not previously exist in \(initialDocCArchiveVersion)._
+                                        
+                    \(additionLinks)
+                    
+                    
+                    #### Removals
+                    _Old symbols that existed in \(initialDocCArchiveVersion) that no longer exist in \(newerDocCArchiveVersion)._
+                                        
+                    \(removalLinks)
+                    
+                    """
+            ]
+        }
+        
+        // MARK: - Command Line Options & Arguments
+        
+        @Argument(
+            help: ArgumentHelp(
+                "The version of the initial DocC Archive to be compared.",
+                valueName: "initialDocCArchiveVersion"))
+        var initialDocCArchiveVersion: String
+        
+        @Argument(
+            help: ArgumentHelp(
+                "The path to the initial DocC Archive to be compared.",
+                valueName: "initialDocCArchive"),
+            transform: URL.init(fileURLWithPath:))
+        var initialDocCArchivePath: URL
```

Comment:
AFAICT the versions are only passed so that they can be included in prose in the generated markup. In my opinion, this extra _required_ information doesn't carry its weight. It's as easy for the developer to fill in the version numbers in the generated markup as it is to provide them to the CLI and it's a lot more explicit.

Reply:
Makes sense to me! I changed these to be optional values instead.